cli/parse_args.ts

```diff
@@ -618,11 +619,13 @@ export function parseArgs<
 
   for (let i = 0; i < args.length; i++) {
     const arg = args[i];
+    assert(typeof arg !== "undefined");
```

Comment:
```suggestion
    assertEquals(arg, undefined);
```
As far as I can see, ditto to other instances of `assert()`.

Reply:
In order to do this and get proper type narrowing, I need to update the assert functions:

from:
```typescript
export function assertNotEquals<T>(actual: T, expected: T, msg?: string): void { /**/ }
```
to
```typescript
export function assertNotEquals<T, E>(
  actual: T,
  expected: E,
  msg?: string,
): asserts actual is Exclude<T, E> { /**/ }
```

Are you happy for me to make this change? Would you like it in this PR or another?

By the way, I could make E extend T (`<T, E extends T>`) but without noUncheckedIndexedAccess we would get errors when writing `assertNotEquals(m[1], undefined);` because undefined wouldn't be part of the type.

Reply:
We could also consider an `assertDefined`:

```
export function assertDefined<T>(
  actual: T,
  msg?: string,
): asserts actual is Exclude<T, null | undefined> { /**/ }
```

Either checking for `null` and `undefined` or just `undefined`.

Reply:
Wow. Crazy that `assertNotEquals()` isn't an assertion signature. Yes please, a separate PR for this would be great!

Reply:
On further investigation, I don't think we need `assertDefined` as it seems we already have `assertExists` that checks for null and undefined. I'll use that for this PR.

As for adding assertion signatures to other assertions, I'll still create the PR but I've discovered that its required to do `<T, E extends T>`. This should be a backwards compatible change. Also it shouldn't be a problem for this PR because we can just use `assertExists` instead.

Reply:
Updated this PR to use assertExists